### PR TITLE
Add content negotiation

### PR DESCRIPTION
**Pull request by sinhaashish**
Client only parses when the content-type of  response is application/xml
Fixes https://github.com/minio/minio-java/issues/718

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -1138,7 +1138,10 @@ private HttpResponse executeReq(Method method, String region, String bucketName,
           errorXml = scanner.next();
         }
 
-        errorResponse = new ErrorResponse(new StringReader(errorXml));
+        if ("application/xml".equals(response.headers().get("content-type"))) {
+          errorResponse = new ErrorResponse(new StringReader(errorXml));
+        }
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: now you are swallowing non-xml response.  you would need to throw an exception about non-xml response received.

Comment (sinhaashish) [reply api/src/main/java/io/minio/MinioClient.java]: Done.

**Comment by sinhaashish** [comment]
@balamurugana @ebozduman  PTAL

File: api/src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -1138,7 +1138,12 @@ private HttpResponse executeReq(Method method, String region, String bucketName,
           errorXml = scanner.next();
         }
 
-        errorResponse = new ErrorResponse(new StringReader(errorXml));
+        if ("application/xml".equals(response.headers().get("content-type"))) {
+          errorResponse = new ErrorResponse(new StringReader(errorXml));
+        } else {
+          errorResponse = new ErrorResponse("Non-Xml Response Received");
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: This is incorrect usage or addition. Each class and error has its own purpose. You would need a new exception `InvalidResponseException` like `NoResponseException`.

Go through javadoc http://minio.github.io/minio-java/ completely.

File: api/src/main/java/io/minio/messages/ErrorResponse.java [hunk 3]
```diff
@@ -64,6 +64,15 @@ public ErrorResponse(Reader reader) throws IOException, XmlPullParserException {
     this.parseXml(reader);
   }
 
+  /**
+   * Constructs a new ErrorResponse object in case of non-xml response.
+   */
+  public ErrorResponse(String message) throws IOException, XmlPullParserException {
```
Comment (balamurugana) [inline api/src/main/java/io/minio/messages/ErrorResponse.java]: This constructor is not used and not required.

Comment (sinhaashish) [reply api/src/main/java/io/minio/messages/ErrorResponse.java]: Removed.

File: api/src/main/java/io/minio/messages/ErrorResponse.java [hunk 4]
```diff
@@ -64,7 +64,6 @@ public ErrorResponse(Reader reader) throws IOException, XmlPullParserException {
     this.parseXml(reader);
   }
 
-
```
Comment (balamurugana) [inline api/src/main/java/io/minio/messages/ErrorResponse.java]: Make zero diff in this file.

File: api/src/main/java/io/minio/MinioClient.java [hunk 5]
```diff
@@ -1138,7 +1139,12 @@ private HttpResponse executeReq(Method method, String region, String bucketName,
           errorXml = scanner.next();
         }
 
-        errorResponse = new ErrorResponse(new StringReader(errorXml));
+        if ("application/xml".equals(response.headers().get("content-type"))) {
+          errorResponse = new ErrorResponse(new StringReader(errorXml));
+        } else {
+          throw new InvalidResponseException();
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: Do a negative logic and you don't need `else` part.

File: api/src/main/java/io/minio/messages/ErrorResponse.java (hunk 4 above)
Comment (sinhaashish) [reply api/src/main/java/io/minio/messages/ErrorResponse.java]: done.

File: api/src/main/java/io/minio/MinioClient.java (hunk 5 above)
Comment (sinhaashish) [reply api/src/main/java/io/minio/MinioClient.java]: Modified.

**Pull request merged**